routing-daemon/lib/openshift/routing/controllers/asynchronous.rb

```diff
@@ -342,13 +342,13 @@ def create_monitor monitor_name, path, up_code, type, interval, timeout
       monitors.push monitor_name
     end
 
-    def delete_monitor monitor_name, pool_name=nil
+    def delete_monitor monitor_name, pool_name=nil, type
       raise LBControllerException.new "Monitor not found: #{monitor_name}" unless monitors.include? monitor_name
 
       # :delete_monitor blocks
       # if the corresponding pool is being deleted (if one is specified) or
       # if the monitor is being created.
-      queue_op Operation.new(:delete_monitor, [monitor_name]), @ops.select {|op| (op.type == :create_monitor && op.operands[0] == monitor_name) || (pool_name && op.type == :delete_pool && op.operands[0] == pool_name)}
+      queue_op Operation.new(:delete_monitor, [monitor_name, type]), @ops.select {|op| (op.type == :create_monitor && op.operands[0] == monitor_name) || (pool_name && op.type == :delete_pool && op.operands[0] == pool_name)}
```

Comment:
Just to close this one out, we talked on IRC and we don't need to add the type check here in the selector since the monitory type is specified in a config file and won't be altered at runtime.